substrate/frame/society/src/lib.rs

```diff
@@ -1382,6 +1388,66 @@ pub mod pallet {
 			// if backend == 0 { return Err(Error::<T, I>::NoVotes.into()); };
 			Ok(Pays::No.into())
 		}
+
+		/// Poke / reconsider the deposit reserved when bidding.
+		///
+		/// The dispatch origin for this call must be _Signed_ and must be the bidder.
+		///
+		/// The transaction fee is waived if the deposit is changed after poking/reconsideration.
+		///
+		/// Emits `DepositPoked` if successful.
+		#[pallet::call_index(20)]
+		#[pallet::weight(T::WeightInfo::poke_deposit())]
+		pub fn poke_deposit(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			// Get current bids and find the bidder's bid
+			let mut bids = Bids::<T, I>::get();
+			let bid_position =
+				bids.iter().position(|bid| bid.who == who).ok_or(Error::<T, I>::NotBidder)?;
+
+			let bid = &mut bids[bid_position];
+
+			// Only handle deposit bids
+			let old_deposit = match &bid.kind {
+				BidKind::Deposit(amount) => *amount,
+				_ => return Err(Error::<T, I>::NotBidder.into()),
```

Comment:
```suggestion
				_ => return Err(Error::<T, I>::NoDeposit.into()),
```
We already checked that they are the bidder?

Reply:
Yes but there are 2 kinds of bids right. So the `BidKind::Vouch` isn't really a `bid` to join a society. I found the lingo a bit confusing in this pallet as vouching and bidding are both treated as bids whereas `bid` is defined as `an offer of a price`. Since there is no deposit associated with `Vouch`, it didn't seem to me like a `bid` so I returned the error `NotBidder` implying that the account trying to execute the `poke_deposit` extrinsic is not really a bidder but a voucher for a bidder.

I might be overthinking this a bit but curious to know what you think @ggwpez @kianenigma 

This PR has been in review for a long time now so I will just change it to `NoDeposit` for now to merge :)